Runtime error viewing kubernetes.io/dockerconfigjson secret

```
angular.js:14199 Error: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.
    at secrets.js:48
    at lodash.js:4944
    at baseForOwn (lodash.js:3001)
    at lodash.js:4913
    at Function.forEach (lodash.js:9359)
    at decodeDockerconfigjson (secrets.js:47)
    at secrets.js:66
    at lodash.js:13379
    at lodash.js:4944
    at baseForOwn (lodash.js:3001)
```

![screen shot 2017-09-12 at 2 02 58 pm](https://user-images.githubusercontent.com/1167259/30341178-195d4ed8-97c3-11e7-929c-3ade4830c245.png)

--- a/app/scripts/services/secrets.js
+++ b/app/scripts/services/secrets.js
@@ -29,10 +29,12 @@ angular.module("openshiftConsole")
     };
 
     var decodeDockercfg = function(encodedData) {
-      var decodedSecretData = {};
+      var decodedSecretData = {
+        auths: {}
+      };
       var decodedData = JSON.parse(window.atob(encodedData));
       _.each(decodedData, function(data, serverName) {
-        decodedSecretData[serverName] = {
+        decodedSecretData.auths[serverName] = {
           username: data.username,
           password: data.password,
           email: data.email
@@ -42,16 +44,28 @@ angular.module("openshiftConsole")
     };
 
     var decodeDockerconfigjson = function(encodedData) {
-      var decodedSecretData = {};
+      var decodedSecretData = {
+        auths: {}
+      };
       var decodedData = JSON.parse(window.atob(encodedData));
       _.each(decodedData.auths, function(data, serverName) {
+        if (!data.auth) {
+          decodedSecretData.auths[serverName] = data;
+          return;
+        }
+
         var usernamePassword = window.atob(data.auth).split(":");
-        decodedSecretData[serverName] = {
+        decodedSecretData.auths[serverName] = {
           username: usernamePassword[0],
           password: usernamePassword[1],
           email: data.email
         };
       });
+
+      if (decodedData.credsStore) {
+        decodedSecretData.credsStore = decodedData.credsStore;
+      }
+
       return decodedSecretData;
     };
 

--- a/app/views/_config-file-params.html
+++ b/app/views/_config-file-params.html
@@ -1,12 +1,24 @@
-<div ng-repeat="(serverName, data) in secretData" class="image-source-item">
+<div ng-repeat="(serverName, data) in secretData.auths" class="image-source-item">
   <h3>{{serverName}}</h3>
-  <dt>username</dt>
-  <dd class="word-break">{{data.username}}</dd>
-  <dt>password</dt>
-  <dd ng-if="view.showSecret">
-    <copy-to-clipboard clipboard-text="data.password" display-wide="true"></copy-to-clipboard>
+  <dt ng-if-start="data.username">username</dt>
+  <dd ng-if-end class="word-break">{{data.username}}</dd>
+  <dt ng-if-start="data.password">password</dt>
+  <dd ng-if-end>
+    <span ng-if="view.showSecret">
+      <copy-to-clipboard clipboard-text="data.password" display-wide="true"></copy-to-clipboard>
+    </span>
+    <span ng-if="!view.showSecret">*****</span>
   </dd>
-  <dd ng-if="!view.showSecret">*****</dd>
-  <dt>email</dt>
-  <dd class="word-break">{{data.email}}</dd>
+  <dt ng-if-start="data.email">email</dt>
+  <dd ng-if-end class="word-break">{{data.email}}</dd>
+  <div ng-if="!data.username && !data.password && !data.email">
+    No username and password.
+  </div>
+</div>
+<h3 ng-if-start="secretData.credsStore">Credentials Store</h3>
+<div ng-if-end>
+  <span ng-if="view.showSecret">
+    <copy-to-clipboard clipboard-text="secretData.credsStore" display-wide="true"></copy-to-clipboard>
+  </span>
+  <span ng-if="!view.showSecret">*****</span>
 </div>

